File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.java
class Solution {
    public int countKConstraintSubstrings(String s, int k) {
        int zero = 0;
        int one = 0;
        int count = 0;

        for (int i = 0; i < s.length(); i++) {
            if(s.charAt(i) - '0' == 0) zero++;
            else one++;

            int localZero = zero;
            int localOne = one;
            for (int j = 0; j <= i; j++) {

                if(Math.min(localZero, localOne) <= k) count++;

                if(s.charAt(j) - '0' == 0) localZero--;
                else localOne--;
            }
        }
        return count;  
    }
}

File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.py
class Solution:
    def countKConstraintSubstrings(self, s: str, k: int) -> int:
        zero = 0
        one = 0
        count = 0
        for i in range(len(s)):
            if ord(s[i]) - ord('0') == 0:
                zero += 1
            else:
                one += 1

            localZero = zero
            localOne = one
            for j in range(i+1):

                if min(localZero, localOne) <= k:
                    count += 1

                if ord(s[j]) - ord('0') == 0:
                    localZero -= 1
                else:
                    localOne -= 1
        return count